### PR TITLE
Updated remoting listener extensions to support exception serialization

### PR DESCRIPTION
**Pull request by WhitWaldo**
The latest guidance on [exception serialization](https://learn.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-exception-serialization), provides examples indicating how the instance and replica listeners need to be updated to accommodate the new serialization settings.

Upon reading these, I realized that I use a provided extension method to configure these listeners in my own code and that this serialization change was not yet realized, so I now offer this PR to resolve this. 

I've added an overload to each extension method (`CreateServiceRemotingInstanceListener()` and `CreateServiceRemotingReplicaListener()`) to allow the developer to pass in an array of `IExceptionConvertor` for any custom exceptions they may wish to accommodate. By default, the extension provides an empty array in order to avoid necessitating any developer change to use this additional functionality. 

In order that this might be consumed without incurring any breaking changes to remoting services, I've added an overload that matches the original signature and provides an empty array of `IExceptionConvertor` to the new overload. All V1 remoting is left untouched as this change is applicable only to V2.

Rider confirms that this code matches code standard requirements and the project builds locally as expected.  Happy to make any changes as required by your team. Thank you for the consideration!

### CODE REVIEW

**Comment by yashagarwal23** [comment]
Thanks for the contribution @WhitWaldo. We are reviewing the PR.

**Review by yashagarwal23 — CHANGES_REQUESTED** [review]

File: src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs [hunk 1]
```diff
@@ -72,14 +73,32 @@ public static class ServiceRemotingExtensions
...
+        /// </summary>
+        /// <typeparam name="TStatefulService">Type constraint on the service implementation. The service implementation must
+        /// derive from <see cref="Microsoft.ServiceFabric.Services.Runtime.StatefulServiceBase"/> and implement one or more
+        /// interfaces that derive from <see cref="Microsoft.ServiceFabric.Services.Remoting.IService"/> interface.</typeparam>
+        /// <param name="serviceImplementation">A stateful service implementation.</param>
+        /// <param name="exceptionConvertors">An array of <see cref="V2.Runtime.IExceptionConvertor"/> to use in serializing and deserializing exceptions.</param>
+        /// <returns>A <see cref="IServiceRemotingListener"/> communication
+        /// listener that remotes the interfaces deriving from <see cref="Microsoft.ServiceFabric.Services.Remoting.IService"/> interface.</returns>
+        public static IEnumerable<ServiceReplicaListener> CreateServiceRemotingReplicaListeners<TStatefulService>(
+            this TStatefulService serviceImplementation, V2.Runtime.IExceptionConvertor[] exceptionConvertors)
```
Comment (yashagarwal23) [inline src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs:93]: We should also have an argument for ExceptionSerializationTechnique in CreateServiceRemotingReplicaListeners/CreateServiceRemotingInstanceListeners alongside the IExceptionConvertor[] exceptionConvertors
The reason for this is that we need to keep BinaryFormatter as the default exception serialisation technique until it is deprecated. If the users do not provide any IExceptionConvertor, we then cant just set the exception serialisation technique to DCS.

If you have a better solution for this, please feel free to try that.

Comment (WhitWaldo) [reply src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs:96]: Approached this in the latest commit by having two extension methods available to the user for either the stateful or stateless instance listeners. One accepts an array of IExceptionConvertor. If this is provided, the user clearly intends to use DSC, so it's configured accordingly. The other allows the user to specify either serialization technique without this IExceptionConvertor array and defaults to the BinaryFormatter (so as not to break existing consumers). Either approach resolves to an extension marked as internal so the user cannot pass arguments to it that wouldn't make sense (an array of IExceptionConvertors + use BinaryFormatting).

File: src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs [hunk 2]
```diff
@@ -90,18 +109,20 @@ public static class ServiceRemotingExtensions
...
                 {
-                    var listeners = provider.CreateServiceRemotingListeners();
-                    foreach (var kvp in listeners)
-                    {
-                        serviceReplicaListeners.Add(new ServiceReplicaListener(
-                            t =>
+                    var listener = new FabricTransportServiceRemotingListener(
+                        serviceImplementation.Context,
+                        impl,
+                        new FabricTransportRemotingListenerSettings
```
[the remark this answers is not in the document]
Comment (WhitWaldo) [reply src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs]: I added `UseWrappedMessage = true` and pull the default settings per the example provided.

File: src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs [hunk 3]
```diff
@@ -7,8 +7,9 @@ namespace Microsoft.ServiceFabric.Services.Remoting.Runtime
 {
     using System;
     using System.Collections.Generic;
-    using System.Fabric;
```
[the remark this answers is not in the document]
Comment (WhitWaldo) [reply src/Microsoft.ServiceFabric.Services.Remoting/Runtime/ServiceRemotingExtensions.cs:10]: Restored this using

**Comment by WhitWaldo** [comment]
@yashagarwal23 Apologies for the delay in circling back here. I've applied the changes requested. Please let me know if you'd like any others made.

**Comment by olegsych** [comment]
Abandoned

**Comment by WhitWaldo** [comment]
@olegsych I haven't abandoned this - I'm happy to make any other changes the team wants to move this forward, but since there's no community call any longer, I don't really have a good mechanism to draw any attention here.

**Pull request closed without merging**